### ISSUE TITLE
editor.renderWhitespace should have option for selection+trailing

### ISSUE THREAD

**Issue by tigerhawkvok** [microsoft/vscode#106697]
Right now you're limited to picking between `selection` (old) or `trailing` (new), but if you want both behaviours you're out of luck.

There should be an option allowing both to be true.

**Comment by amomchilov** [comment]
Yeah, it was a mistake IMO to model these as a single value. Should be a series of checkboxes, e.g:

* Render trailing whitespace
* Render leading whitespace
* Render all whitespace in selection
* And one for the "boundary" option, which I frankly don't even understand.

**Comment by hika-ru** [comment]
Are there any news on this one? I'd love to have it fixed for use in markdown editing 🙂.

**Comment by tuxcomputers** [comment]
> Yeah, it was a mistake IMO to model these as a single value. Should be a series of checkboxes, e.g:
> 
>     * Render trailing whitespace
> 
>     * Render leading whitespace
> 
>     * Render all whitespace in selection
> 
>     * And one for the "boundary" option, which I frankly don't even understand.

Boundary is trailing + leading + more than single space within the text. This is boundary, the red arrow is the whitespace rendering for more than single space
![Screenshot_1404](https://user-images.githubusercontent.com/26516524/168698013-8bab5cb4-f423-4ada-99f8-b4fe20ce8514.png)

If it was a series of checkboxes it would solve my request, I want boudary without the last one, I could just tick leading and trailing.

**Comment by laurentlb** [comment]
I like the suggestions above, but I'm not certain the VS Code team would be happy with a bunch of checkboxes. As an alternative, we could:

1. Remove the `Selection` option from the `Editor: Render Whitespace` enum
    * New options could be added if needed, e.g. `Leading + Trailing`
2. Introduce a separate setting, e.g. "Render whitespace when selected"

I feel like the "selected" option is distinct enough that it deserves its own setting.

Would a PR be accepted for this?

**Comment by injust** [comment]
Bumping this as it seems like it's been forgotten

**Comment by torzod** [comment]
any updates on this feature? would be very useful for me and evidently many others, thanks!

**Comment by ka-iden** [comment]
Bump ;)

**Comment by Jouklik** [comment]
Just found out that two spaces followed by line break in markdown translate into `<br>`. 

So, I thought let's enable **trailing white space**. It would be super useful **along with** super useful **selection white space.** Well, apparently **NO.**

**Comment by maxnevans** [comment]
Having all or boundary whitespaces is overwhelming for me. Though, I'd love to see trailing whitespaces as well as all the whitespaces in selected text. So, yeah, waiting for this feature as well.

**Comment by palait** [comment]
Any updates on this?
